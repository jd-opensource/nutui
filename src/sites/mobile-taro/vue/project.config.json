{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
