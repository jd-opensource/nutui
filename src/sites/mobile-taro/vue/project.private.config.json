{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}