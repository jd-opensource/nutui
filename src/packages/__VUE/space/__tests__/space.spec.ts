import { mount } from '@vue/test-utils';
import Space from '../index.vue';
import NutButton from '../../button/index.vue';
import { h } from 'vue';

const prefixCls = 'nut-space';

test('space: should match snapshot', () => {
  const wrapper = mount(Space, {
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('space: with props direction', () => {
  const wrapper = mount(Space, {
    props: {
      direction: 'vertical'
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.classes()).toContain(`${prefixCls}-vertical`);
});

test('space: with props align', () => {
  const wrapper = mount(Space, {
    props: {
      align: 'end'
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.classes()).toContain(`${prefixCls}-align-end`);
});

test('space: with props justify', () => {
  const wrapper = mount(Space, {
    props: {
      justify: 'center'
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.classes()).toContain(`${prefixCls}-justify-center`);
});

test('space: with props wrap', () => {
  const wrapper = mount(Space, {
    props: {
      wrap: true
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.classes()).toContain(`${prefixCls}-wrap`);
});

test('space: with props fill', () => {
  const wrapper = mount(Space, {
    props: {
      fill: true
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  expect(wrapper.classes()).toContain(`${prefixCls}-fill`);
});

test('space: with props gutter number', () => {
  const wrapper = mount(Space, {
    props: {
      gutter: 20
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  const nutSpaceItem = wrapper.find('.nut-space-item');
  expect(nutSpaceItem.attributes('style')).toBe('margin-right: 20px;');
});

test('space: with props gutter string', () => {
  const wrapper = mount(Space, {
    props: {
      gutter: '1rem'
    },
    slots: {
      default: [h(NutButton, () => '按钮'), h(NutButton, () => '按钮'), h(NutButton, () => '按钮')]
    }
  });
  const nutSpaceItem = wrapper.find('.nut-space-item');
  expect(nutSpaceItem.attributes('style')).toBe('margin-right: 1rem;');
});
