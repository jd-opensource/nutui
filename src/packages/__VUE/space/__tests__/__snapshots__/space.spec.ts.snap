// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`space: should match snapshot 1`] = `
"<div class="nut-space nut-space-horizontal">
  <div class="nut-space-item">
    <view class="nut-button nut-button--default nut-button--normal nut-button--round">
      <view class="nut-button__wrap">
        <!--v-if-->
        <!--v-if-->
        <view class="">按钮</view>
      </view>
    </view>
  </div>
  <div class="nut-space-item">
    <view class="nut-button nut-button--default nut-button--normal nut-button--round">
      <view class="nut-button__wrap">
        <!--v-if-->
        <!--v-if-->
        <view class="">按钮</view>
      </view>
    </view>
  </div>
  <div class="nut-space-item">
    <view class="nut-button nut-button--default nut-button--normal nut-button--round">
      <view class="nut-button__wrap">
        <!--v-if-->
        <!--v-if-->
        <view class="">按钮</view>
      </view>
    </view>
  </div>
</div>"
`;
